extensions/binding/org.eclipse.smarthome.binding.meteoblue/ESH-INF/thing/thing-types.xml

```diff
@@ -110,12 +104,12 @@
 			<channel id="snowFall" typeId="snowFall" />
 			<channel id="cardinalWindDirection" typeId="cardinalWindDirection" />
 			<channel id="windDirection" typeId="windDirection" />
```

Comment:
So, for this, I would need to remove the entire `<channel-type>` block for each affected item?

Delete this:

	<channel-type id="pressureMax">
		<item-type>Number:Pressure</item-type>
		<label>High Pressure</label>
		<description>High Pressure</description>
		<category>Pressure</category>
		<state readOnly="true" pattern="%.2f %unit%" />
	</channel-type>

And change this:

	<channel id="maxSeaLevelPressure" typeId="pressureMax" />

To this?

	<channel id="maxSeaLevelPressure" typeId="barometric-pressure" />

Reply:
Almost, you have to remove the channel-type definition and change your reference to

```
<channel id="maxSeaLevelPressure" typeId="system.barometric-pressure" />
```

Reply:
After doing this, I get warnings for the system.outdoor-temperature type:

````
07:58:48.744 [WARN ] [core.internal.thing.ThingTypeResource] - Cannot find channel type: system:outdoor-temperature
````

Reply:
And in addition, it appears none of these types work.

![image](https://user-images.githubusercontent.com/9916473/43045509-d288a00a-8d87-11e8-984b-dd0099a83a92.png)

Reply:
Do you have the recent commit https://github.com/eclipse/smarthome/commit/7bb5ac0474a2736ee92aabda6307fd4e390b4bd9 from the master branch in your branch? If not, please rebase yours onto the current master and try again.

Reply:
I just realized why it's not working. I'm running the 2.3.0 OH2 release, which doesn't have this commit.
So I can either add these type changes without testing them, or leave them out.

Reply:
Where are you developing your binding? Inside the ESH IDE? Or an openHAB IDE? Since you created a PR against ESH directly, I am assuming the former. In there you can easily rebase your branch and start ESH with your bundle, right?

If you are inside the openHAB IDE you can edit the run configuration, disable the `org.eclipse.smarthome.core.thing` bundle from the target platform and select it from the checked out project in your work space which should then of course be based on the current master from ESH.

Does that help?

Reply:
If you're referring to [this](https://docs.openhab.org/developers/development/ide.html), I've never been able to get a functional workspace from it. Thousands of unresolvable bogus errors.

So I've never been able to use an IDE for debugging...

Reply:
In your case it would even be sufficient to use the ESH IDE: https://www.eclipse.org/smarthome/documentation/development/ide.html

I have followed these steps just last week to install Photon and it worked with just one small issue: After following all steps I had to manually clean the projects with errors and let eclipse rebuild them twice or maybe even 3 times. In case Paper UI doesnt work out of the box, just run "PaperUI Setup.launch" from the paperUi project again.

Reply:
I'll give this a shot and see what I get...

Reply:
So not good.

After running all of the steps, I reach the one that says "Your workspace should now fully compile", and as usual, it does not:

![image](https://user-images.githubusercontent.com/9916473/43199118-eed7c598-8fde-11e8-82f2-1371a4b1457b.png)

![image](https://user-images.githubusercontent.com/9916473/43199146-0961fbf4-8fdf-11e8-815f-c80662791948.png)

Reply:
Please see https://github.com/eclipse/smarthome/pull/4821#discussion_r205007871

I had to clean the projects with errors several times too, until they finally compiled.

Reply:
How many dozens of cleans are we going to say is appropriate?
I've done several and the number of errors never declines below 286.

Reply:
Strange... For me it worked. What you can try is to open the `org.eclipse.smarthome.model.codegen` project and let the `Generate All Models.launch` script run again (ignore that projects might have errors). And afterwards clean the projects with errors one more time.

If this works, we probably should update our documentation, that after cleaning and rebuild these issues can be solved by running the model generation again.

Reply:
If I right-click any single project and try to build it, I get this:

![image](https://user-images.githubusercontent.com/9916473/43456340-2608dec4-9491-11e8-92a0-23e8546d2f42.png)

After I closed all the projects and opend the `codegen` project, it built successfully. I then ran the `Generate All Models.launch` script. It generated a number of errors like this:

````
Could not load class: org.eclipse.core.runtime.OperationCanceledException
Add org.eclipse.equinox.common to the class path.
````

I then reopened all the projects. After rebuilding, the number of errors is now 379.
Wrong direction.

Reply:
I am sorry to say that I am out of ideas for now...

Any chance of using the latest OH 2.4.0-SNAPSHOT for testing, since you said you have 2.3.0?

Otherwise I would say that you leave out these new system channels and someone can come up with a follow-up PR.

Reply:
At this point, I'm going to advocate for leaving them out.
If someone thinks they're useful, they can add them later.